glib/src/main_context_channel.rs

```diff
@@ -66,6 +66,7 @@ impl<T> Clone for Channel<T> {
 }
 
 impl<T> Channel<T> {
+    #[allow(unknown_lints, clippy::arc_with_non_send_sync)]
```

Comment:
This should get a clippy issue. There's nothing wrong here and it's too strict. We use `T` instead of `T: Send + Sync` here to allow both cases, and there's no other way to allow both cases really.

clippy should probably only warn about this for concrete types.